fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3499,6 +3492,49 @@ public CompletableFuture<IndexBuildState> getIndexBuildStateAsync(Index index) {
         return IndexBuildState.loadIndexBuildStateAsync(this, index);
     }
 
+    /**
+     * Load the indexing type stamp for an index. This stamp contains information about the kind of
+     * index build being used to construct a new index. This method is {@link API.Status#INTERNAL}.
+     *
+     * @param index the index being built
+     * @return the indexing type stamp for the index's current build
+     * @see #saveIndexingTypeStamp(Index, IndexBuildProto.IndexBuildIndexingStamp)
+     */
+    @API(API.Status.INTERNAL)
+    @Nonnull
+    public CompletableFuture<IndexBuildProto.IndexBuildIndexingStamp> loadIndexingTypeStampAsync(Index index) {
+        byte[] stampKey = IndexingBase.indexBuildTypeSubspace(this, index).pack();
```

Comment:
It seems a bit weird that `IndexingBase` continues to manage what the subspace is, but not how to write to it.